presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -649,8 +649,7 @@ public IntPredicate getSpillMask()
 
         public SavedRow(Page page, int position, long joinPositionWithinPartition, boolean currentProbePositionProducedRow, int joinSourcePositions)
         {
-            this.row = page.mask(new int[] {position});
-            this.row.compact();
+            this.row = page.getSingleValuePage(position);
```

Comment:
Can you double check that `mask` + `compact` has the same result as `getSingleValuePage`... specifically that we always end up with a copy.

Reply:
`Page.compact` has `DictionaryBlock`-related logic which I don't understand, so I won't speak about this part.

Other than that, I looked through implementations of `Block. getSingleValueBlock` and everywhere it looks like simplified implementation of `Block.copyRegion`. There is always a copy.